fplc_method_validator.py

```
@@ -0,0 +1,497 @@
+
+        parsed_steps: List[AnyMethodStep] = []
+        try:
+            for i, step_data in enumerate(steps_data):
+                step_type_str = step_data.get('type')
+                if not step_type_str:
+                    raise TemplateError(f"Step {i+1} in template '{name}' is missing 'type'.")
+
+                try:
+                    step_type = StepType(step_type_str)
+                except ValueError:
+                    raise TemplateError(f"Step {i+1} in template '{name}' has an unknown step type: '{step_type_str}'.")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)